src/plugins/data_source/server/data_source_route_handler_context.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// eslint-disable-next-line max-classes-per-file
+import { Logger } from 'src/core/server';
+import { IDataSourceClient } from './client/data_source_client';
+
+class OpenSearchDataSourceRouteHandlerContext {
+  private logger: Logger;
+  constructor(private dataSourceClient: IDataSourceClient, logger: Logger) {
```

Comment:
add dataSourceClient to class field as well to be consistent?

Reply:
I'll remove the logger, to use parameter property here for consistency. 
https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties